lib/syscall_shim/uk_prsyscall.c

```diff
@@ -1224,16 +1224,16 @@ static void pr_syscall(struct uk_streambuf *sb, int fmtf,
 		VPR_SYSCALL(sb, fmtf, syscall_num, args, rc >= 0,
 			    PT_CLONEFLAGS,
 			    PT_VADDR, /* sp */
-			    PT_TID | PT_REF, /* ref to parent tid */
```

Comment:
Looks good. But, `pr_param`'s `PT_REF` seems to **supposedly** handle NULL though? Could the bug be somewhere else instead?

Reply:
Sorry, the description is incorrect as I opened the PR long time after I wrote the patch. The actual problem is that if the caller does not pass `CLONE_CHILD_SETTID` / `CLONE_PARENT_SETTID` the pointers will be invalid, so a check for `NULL` does not suffice. While I was writing that though I came up with a better fix that retains the previous behavior if these flags are set. Will push and update the commit message & cover letter.